cmd/soroban-rpc/internal/methods/get_transactions.go

```diff
@@ -0,0 +1,267 @@
+	//if err != nil {
+	//	return errors.New("could not get latest ledger")
+	//}
+	return nil
+}
+
+type GetTransactionsRequest struct {
+	StartLedger uint32
+	EndLedger   uint32
+	Pagination  *TransactionsPaginationOptions
+}
+
+func (req GetTransactionsRequest) isValid() error {
+	// Validate the start and end ledger sequence
+	if req.StartLedger < 0 {
+		return errors.New("start ledger cannot be negative")
+	}
+	if req.EndLedger < req.StartLedger {
+		return errors.New("end ledger cannot be less than start ledger")
+	}
+
+	return nil
+}
+
+type TransactionInfo struct {
+	Result           []byte // XDR encoded xdr.TransactionResult
+	Meta             []byte // XDR encoded xdr.TransactionMeta
+	Envelope         []byte // XDR encoded xdr.TransactionEnvelope
+	FeeBump          bool
+	ApplicationOrder int32
+	Successful       bool
+	LedgerSequence   uint
+}
+
+type GetTransactionsResponse struct {
+	Transactions               []TransactionInfo `json:"transactions"`
+	LatestLedger               int64             `json:"latestLedger"`
+	LatestLedgerCloseTimestamp int64             `json:"latestLedgerCloseTimestamp"`
+	Pagination                 *TransactionsPaginationOptions
+}
+
+type transactionsRPCHandler struct {
+	ledgerReader      db.LedgerReader
+	ledgerEntryReader db.LedgerEntryReader
+	maxLimit          uint
+	defaultLimit      uint
+	logger            *log.Entry
+	networkPassphrase string
+}
+
+func (h transactionsRPCHandler) getTransactionsByLedgerSequence(ctx context.Context, request GetTransactionsRequest) (GetTransactionsResponse, error) {
+	err := request.isValid()
+	if err != nil {
+		return GetTransactionsResponse{}, &jrpc2.Error{
+			Code:    jrpc2.InvalidParams,
+			Message: err.Error(),
+		}
+	}
+
+	start := transactions.NewCursor(request.StartLedger, 0, 0)
+	endLedger, found, err := h.ledgerReader.GetLedger(ctx, request.EndLedger)
+	if err != nil || !found {
+		return GetTransactionsResponse{}, &jrpc2.Error{
+			Code:    jrpc2.InvalidParams,
+			Message: err.Error(),
+		}
+	}
+	end := transactions.NewCursor(request.EndLedger, uint32(endLedger.CountTransactions()), 0)
+
+	// Move start to pagination cursor
+	limit := h.defaultLimit
+	if request.Pagination != nil {
+		if request.Pagination.Cursor != nil {
+			start = *request.Pagination.Cursor
+			// increment event index because, when paginating,
+			// we start with the item right after the cursor
+			start.TxIdx++
+		}
+		if request.Pagination.Limit > 0 {
+			limit = request.Pagination.Limit
+		}
+	}
+
+	// Validate the range
+	ledgerRange := Range{
+		Start: start,
+		End:   end,
+	}
+	err = ledgerRange.isValid(ctx, h.ledgerReader, h.ledgerEntryReader)
+	if err != nil {
+		return GetTransactionsResponse{}, &jrpc2.Error{
+			Code:    jrpc2.InvalidParams,
+			Message: err.Error(),
+		}
+	}
+
+	// Get all ledgers within the range from db
+	ledgers, err := h.ledgerReader.GetLedgers(ctx, ledgerRange.Start.LedgerSequence, ledgerRange.End.LedgerSequence)
```

Comment:
@Shaptic Yes I just quickly added this so that I can do a quick implementation. This is something we should discuss though - doing it one ledger at a time vs in batches.

Reply:
Actually I will start with just reading one ledger at a time since the pagination limit will be on transactions which would be hit sooner than we finish off all the ledgers in the range.